src/m365/spo/commands/applicationcustomizer/applicationcustomizer-add.ts

```diff
@@ -0,0 +1,109 @@
+import * as spoCustomActionAddCommand from '../customaction/customaction-add';
+import Command from '../../../../Command';
+import { Cli } from '../../../../cli/Cli';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  title: string;
+  webUrl: string;
+  clientSideComponentId: string;
+  clientSideComponentProperties?: string;
+}
+
+class SpoApplicationCustomizerAddCommand extends SpoCommand {
+  public get name(): string {
+    return commands.APPLICATIONCUSTOMIZER_ADD;
+  }
+
+  public get description(): string {
+    return 'Add an application customizer to a site.';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-t, --title <title>'
+      },
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-i, --clientSideComponentId <clientSideComponentId>'
+      },
+      {
+        option: '--clientSideComponentProperties [clientSideComponentProperties]'
+      },
+    );
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        clientSideComponentProperties: typeof args.options.clientSideComponentProperties !== 'undefined'
+      });
+    });
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.webUrl) {
+          const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+          if (isValidSharePointUrl !== true) {
+            return isValidSharePointUrl;
+          }
+        }
+
+        if (!validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        if (args.options.clientSideComponentProperties) {
+          try {
+            JSON.parse(args.options.clientSideComponentProperties);
+          }
+          catch (e) {
+            return `An error has occurred while parsing clientSideComponentProperties: ${e}`;
+          }
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Adding application customizer with title '${args.options.title}' and clientSideComponentId '${args.options.clientSideComponentId}' to the site`);
+    }
+
+    const options: spoCustomActionAddCommandOptions = {
+      webUrl: args.options.webUrl,
+      name: args.options.title,
+      title: args.options.title,
+      clientSideComponentId: args.options.clientSideComponentId,
+      clientSideComponentProperties: args.options.clientSideComponentProperties || '',
+      location: 'ClientSideExtension.ApplicationCustomizer'
+    };
```

Comment:
We should pass verbose and debug options here, so that when user requests either output mode, we show that information. We should also pass output so that the command will return output in the specified output mode.

When I add these settings, I'm seeing the `adds application customizer to a specific site while specifying clientSideComponentProperties` test failing with `Cannot read properties of undefined (reading 'ClientSideComponentId')`. Could you please have a look at what's wrong?

Reply:
Apparently, the command logs the output of the customaction add response when passing verbose. This is why. It has been solved now.